pkg/reconciler/pipeline/controller.go

```diff
@@ -103,6 +105,7 @@ func (r *pipelineReconciler) reconcile(ctx context.Context, pl *dfv1.Pipeline) (
 				safeToDelete, err := r.safeToDelete(ctx, pl)
 				if err != nil {
 					log.Errorw("Failed to check if it's safe to delete the pipeline", zap.Error(err))
+					r.recorder.Event(pl, corev1.EventTypeWarning, "ReconcilePipelineFailed", "Failed to check if it's safe to delete the pipeline")
 					return ctrl.Result{}, err
```

Comment:
I see the underlying errors aren't being included in the message - maybe it would be helpful to include them? (unless there's some reason we don't want the message to be too verbose) If I glance at the use of Events in the Argo Workflows code, at least in certain places I see we are including a full error.

Reply:
This is true, there is a mix of it in Argo Workflows. There is a fear that the events would be too verbose - these are supposed to be readable.

@KeranYang thoughts?

Reply:
for the sake of eliminating redundancy, you could consider doing an `fmt.Sprintf()` to format the string which becomes both the error to log and the event message to emit

Reply:
There is also the event function that does formatting:
```
// example
r.recorder.Eventf(pl, corev1.EventTypeWarning, "ReconcilePipelineFailed", "Failed to check if it's safe to delete the pipeline: %v", err)
```

Reply:
I am ok with including the full error.

Reply:
I was thinking we could remove the redundancy of writing out the error message twice? Like:
`fmt.Sprintf("Failed to check if it's safe to delete the pipeline: %v", err.Error())` and then apply it to both the log and the event?

Reply:
+1